docs/template-context.md

```diff
@@ -0,0 +1,41 @@
+---
+title: "Template Context"
+weight: 80
+---
+
+While using the generator tool you may want dynamic values to be populated to your templates and thus be rendered in the output. The generator is able to achieve that using the **template context**.
+The **template context** allows you to access the contents of the [AsyncAPI document](asyncapi-document.md) and inject dynamic values to the template files passed to the asyncAPI CLI during the generation process. The render engine will then display these dynamically assigned values in the output.
+
+## Template context generation process
```

Comment:
```suggestion
## Generation process
```

Since this doc's title is already "template context," I don't think this header needs that phrase included again.